Clarify ST_ShiftLongitude docs

LKajan opened the pull request:
The ST_ShiftLongitude function behave symmetrical with inverse and reverse transformations. Clarify this in the docs.

resolves #5010
https://trac.osgeo.org/postgis/ticket/5010

robe2 commented:
@LKajan can you make changes based on @pramsey input or comment.  Then we can accept this.

LKajan commented:
Thanks @pramsey for your comments!
I changed back to using term shift. I trust your view here being a native speaker.
My reasoning to change the term was that I see we have here just two different representations of a same coordinate point and we just swap the representations back and forth.

In the project do you like more amended commits and force push to the pr or separate commits and squash before merge? I now added these fixes as separate commits.
